src/main/scala/mesosphere/marathon/raml/AppConversion.scala

```diff
@@ -308,6 +303,20 @@ trait AppConversion extends DefaultConversions with ConstraintConversion with En
       ))
       else None
 
+    val unreachableStrategy = service.when(_.hasUnreachableStrategy, _.getUnreachableStrategy.toRaml).orElse(App.DefaultUnreachableStrategy)
+
+    val hasPersistentVolumes = service.hasContainer && service.getContainer.getVolumesList.exists(_.hasPersistent)
+
+    val residency = if (hasPersistentVolumes || service.hasResidency) {
+      unreachableStrategy.flatMap {
+        case raml.UnreachableDisabled(_) => None
```

Comment:
The same here, shouldn't it be `residency.taskLostBehavior = WAIT_FOREVER`, so that we stay backward compatible, until `residency` is removed, not just deprecated?